src/cpp/server/ServerSessionProxy.cpp

```diff
@@ -275,7 +285,40 @@ void handleLocalhostResponse(
       }
       else
       {
-         ptrConnection->writeResponse(response);
+         // fixup bad SparkUI URLs in responses (they use paths hard
+         // coded to the root "/" and we are proxying them behind
+         // a "/p/<port>/" URL)
+         if (boost::algorithm::contains(response.body(),
+                                        "<img src=\"/static/spark-logo"))
+         {
+            // determine url prefix based on how deep we are in the path
+            std::string path = ptrConnection->request().path();
+            size_t slashes = std::count(path.begin(), path.end(), '/');
+            size_t up = slashes - 3;
```

Comment:
IIRC `size_t` is unsigned so if you somehow got < 3 slashes you could wind up with an underflow here.